lib/mintacoin/accounts/crypto/stellar.ex

```diff
@@ -16,28 +16,52 @@ defmodule Mintacoin.Accounts.Stellar do
 
   @impl true
   def create_account(_opts) do
-    {fund_public_key, _fund_private_key} = fund_keypair = fund_key_pair_from_system()
+    fund_info = fund_information()
     {public_key, secret_key} = KeyPair.random()
 
-    source_account = TxBuild.Account.new(fund_public_key)
-    {:ok, seq_num} = Horizon.Accounts.fetch_next_sequence_number(fund_public_key)
-    sequence_number = TxBuild.SequenceNumber.new(seq_num)
-
     operation =
-      TxBuild.CreateAccount.new(
-        destination: public_key,
-        starting_balance: 1
-      )
+      public_key
+      |> build_create_account_operation()
+
+    envelope = build_envelope(fund_info.fund_public_key, operation, fund_info.fund_signature)
+
+    envelope
+    |> execute_transaction(public_key, secret_key)
+  end
 
-    signature = TxBuild.Signature.new(fund_keypair)
+  defp fund_information do
+    {pk, _fund_private_key} = fund_keypair = fund_key_pair_from_system()
 
-    {:ok, envelope} =
-      source_account
-      |> TxBuild.new(sequence_number: sequence_number)
-      |> TxBuild.add_operation(operation)
-      |> TxBuild.sign(signature)
-      |> TxBuild.envelope()
+    signature =
+      fund_keypair
+      |> TxBuild.Signature.new()
+
+    %{fund_public_key: pk, fund_signature: signature}
+  end
+
+  defp build_create_account_operation(destination, starting_balance \\ 1) do
+    TxBuild.CreateAccount.new(
+      destination: destination,
+      starting_balance: starting_balance
+    )
+  end
+
+  defp build_envelope(fund_public_key, operation, signature) do
+    sequence_number =
+      fund_public_key
+      |> Horizon.Accounts.fetch_next_sequence_number()
+      |> TxBuild.SequenceNumber.new()
+
+    source_account = TxBuild.Account.new(fund_public_key)
+
+    source_account
+    |> TxBuild.new(sequence_number: sequence_number)
```

Comment:
@Odraxs You can group this part into the pipes, for example
```suggestion
    fund_public_key
    |> TxBuild.Account.new()
    |> TxBuild.new(sequence_number: sequence_number)
```